### PR TITLE
[FlashScript] version 2.3.0

### PR DESCRIPTION
**Pull request by andyblac**
add support for using IP addresses in manual hosts
add --local-file for support flash local zip files (place zip file in 'tools' folder)
add up time.

`./flash-shelly.py --local-file=firmware.zip shellyrgbw2-xxxxxx.local`

### CODE REVIEW

File: tools/flash-shelly.py [hunk 1]
```diff
@@ -204,44 +243,91 @@ def get_current_version(self): # used when flashing between formware versions.
...
+    if os.path.exists(self.local_file):
+      local_host = socket.gethostname()
+      s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+      s.connect((self.wifi_ip, 80))
+      local_ip = s.getsockname()[0]
+      logger.debug(f"Host IP: {local_ip}")
+      s.close()
+      with zipfile.ZipFile(self.local_file, "r") as zfile:
+        for name in zfile.namelist():
+          if re.search(r'\.json$', name) is not None:
```
Comment (rojer) [inline tools/flash-shelly.py:296]: you need `manifest.json`, not just any .json

Comment (andyblac) [reply tools/flash-shelly.py]: oops ye, https://github.com/mongoose-os-apps/shelly-homekit/pull/492/commits/691edaadb9c2112c54d48669c97308c401e374cd fixes that.

File: tools/flash-shelly.py [hunk 2]
```diff
@@ -293,11 +293,12 @@ def parse_local_file(self):
       s.close()
       with zipfile.ZipFile(self.local_file, "r") as zfile:
         for name in zfile.namelist():
-          if re.search(r'\.json$', name) is not None:
+          if re.search(r'manifest.json$', name) is not None:
```
Comment (rojer) [inline tools/flash-shelly.py:296]: if name.endswith('manifest.json')

Comment (andyblac) [reply tools/flash-shelly.py:296]: not sure is will search subdirectories using that, let me test.

Comment (andyblac) [reply tools/flash-shelly.py]: ok works, done

**Comment by rojer** [comment]
mgd, thanks!

**Comment by andyblac** [comment]
> mgd, thanks!

np, been using alot to help test alpha rgbw2 firmware 😄

**Pull request merged**